atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/TransactionManager.java

```diff
@@ -17,6 +17,17 @@
 
 public interface TransactionManager extends AutoCloseable {
     /**
+     * Used for TransactionManagers that can be initialized asynchronously (i.e. those extending
+     * {@link com.palantir.async.initializer.AsyncInitializer}; other TransactionManagers can keep the default
+     * implementation, and return true (they're trivially fully initialized).
+     *
+     * @return true if and only if the TransactionManager has been fully initialized
```

Comment:
We probably want to add some user-oriented docs on this method (could just steal these from https://github.com/palantir/atlasdb/pull/2441/files)